Jenkinsfile

```diff
@@ -1,209 +1,211 @@
+                    println("Anaconda Path: " + condaPath)
+                    
+                    sh """
+                        # Use the conda cache running on the Jenkins host
+                        conda config --set always_yes True
+                        conda config --set ssl_verify false 
+                    """
+                    
+                    if (lower_label == "centos") {
+                        sh "conda env create --prefix ${condaPath}/envs/isis -f environment_gcc4.yml"
+                    } else {
+                      sh """
+                          conda config --show channels
+                          conda env create --prefix ${condaPath}/envs/isis -f environment.yml
+                      """
+                    }
+                    dir("${env.ISISROOT}") {
+                        try {
+                              env.STAGE_STATUS = "Building ISIS on ${label}"
+                              sh """
+                                  source activate ${condaPath}/envs/isis
+                                  echo `ls ../`
+                                  echo `pwd`
+                                  conda list
+                                  cmake -GNinja ${cmakeFlags.join(' ')} ../isis
+                                  ninja -j4 install
+                              """
+                        }
+                        catch(e) {
+                            build_ok = false
+                            errors.add(env.STAGE_STATUS)
+                            println e.toString()
+                        }
 
-                                        if(build_ok) {
-                                            currentBuild.result = "SUCCESS"
-                                        }
-                                        else {
-                                            currentBuild.result = "FAILURE"
-                                            def comment = "Failed during:\n"
-                                            errors.each {
-                                                comment += "- ${it}\n"
-                                            }
-                                            groovy_utilities.setGitHubBuildStatus(comment)
-                                        }
-                                    }
+                        if (build_ok) {
+                            try{
+                                dir("${env.ISISROOT}") {
+                                    env.STAGE_STATUS = "Running unit tests on ${label}"
+                                    sh """
+                                        source activate ${condaPath}/envs/isis
+                                        ctest -R _unit_ -j4 -VV
+                                    """
                                 }
                             }
+                            catch(e) {
+                                build_ok = false
+                                echo e.toString()
```

Comment:
Why does this not have `errors.add(env.STAGE_STATUS)`?

Reply:
Can all of these catch blocks be a general function? We could add something like conda list, pwd, printenv to help debugging.

Reply:
I think it's fine leaving it as is for now and if we need to change all of these at once in the future then changing it to a function is fine.